### ISSUE TITLE
Getting hung up on launching PyDev debug configuration

### ISSUE THREAD

**Issue by humbienri** [nokia/RED#308]
Hello all,

I've been running into some issues lately but I've been so distracted that I fell like I figure it out and then forget it.

I've recently started doing debugging involving both PyDev for a custom python library that I call from my Robot Resource file, which in turn is called from the Robot test case.  I did manage to successfully set it all up and hit my breakpoints w/i the custom python library at one point.  But once I modify my custom library's constructor to include parameters and inheritance (locustio) then I am stuck again.

I've manage to use python libdoc to manually generate my documentation XML file and in the Project explorer in RED I can see the Keywords of the custom library in the Robot Referenced libraries.

However, in my Resource file, the referencing/importing of the library (Library  customlib) is still highlighting red and claiming the Keyword (not library, but keyword) is expecting 1 argument but 0 are provided.

When I try to launch my Debug Configuration (1st starting the PyDev debugger, then launching the debug configuration, then RED just seems to hang and get stuck running the custom debug configuration I created.

The Progress View/Perspective just shows the green progress bar and never ends.

![_2019Jun03213645](https://user-images.githubusercontent.com/1178569/58847208-c193ce00-8647-11e9-92d0-6444f665c4d7.png)

Any advise would greatly be appreciated.

**Comment by humbienri** [comment]
And my custom lib's constructor is this:
```
class locustfile(TaskSet):

    def __init__(self, TaskSet):
        TaskSet.__init__(self)
```

**Comment by michalanglart** [comment]
> However, in my Resource file, the referencing/importing of the library (Library customlib) is still highlighting red and claiming the Keyword (not library, but keyword) is expecting 1 argument but 0 are provided.

There is a minor problem in the message: it should state that the library constructor is expecting 1 argument. Apart from that it is fine: your library indeed require 1 argument to be provided and your `.robot` source does not provide any in your `Library` import. Robot would report an error too once trying to execute this import.

IMHO there should be a good reason why the library would require arguments. Ask yourself a question: if the user of my library just wants to generate its documentation then should we bother him/her about parameters which are required in order generate this documentation? There are libraries for which there is a very good reason to do this - take for example standard `Remote` library. It takes two parameters: `uri` and `timeout`. While `timeout` is not very important the `uri` is crucial: depending on the value of this parameter the library may have totally different keywords since it would connect to different servers. They made `uri` optional assuming some default host to which the library is connecting however to me it looks like a good candidate to be a required argument. Most of the libraries however can have only optional parameters in constructor and this makes life easier for user who just wants to generate and read the documentation without thinking about parameters required to be passed.

As to the hanging/stucking issue: I'm unable to reproduce given limited information. Are you able to run similar configuration but without without python debugger in both Run and Debug mode (RED debugger only)? Does debugging only robot code works?

**Comment by humbienri** [comment]
Good question, actually I am not able to debug anything at all now.  Even without engaging PyDev debugger.  RED gets hung up on Discovering Libraries and it always seems to hang up on the BuiltIn library.

![_2019Jun04100100](https://user-images.githubusercontent.com/1178569/58890145-b1173e00-86af-11e9-910e-93273cbf873f.png)

It appears that the robot debug session terminates?

**Comment by humbienri** [comment]
I do appreciate your thoughts and insight on whether to have a parameterless constructor or not.  I guess this is over my head though since AFAIK, or can tell, my class has to inherit from the TaskSet class from Locust.

But in order to properly generate the libdoc XML file, I have to put in a constructor there.  I don't even remember how I came up with the TaskSet.__init__() call.  Kinda random.  I can't even explain it.

**Comment by michalanglart** [comment]
I'm not familiar with Locust library however simple examples of classes implementing `TaskSet` included in documentation (https://docs.locust.io/en/stable/writing-a-locustfile.html) do not have constructor defined at all. So, why do you have to do it?

As to this construction:
```
class locustfile(TaskSet):

    def __init__(self, TaskSet):
        TaskSet.__init__()
```

What exactly should this little snippet do? There is a confusion between two different `TaskSet` names: firstly it is used to designate from which class the `locusfile` class is inheriting, secondly same name is used as argument in constructor and then we call another constructor on passed object. Why?

Basically we can construct objects of `locusfile` class by passing any object:

```
a = locusfile(42)
b = locusfile('hello')
c = locusfile([1,2,3])
d= ...
```

and it will all work. What's the point here?

**Comment by humbienri** [comment]
Hello again @michalanglart,

If I try to implement my class locustfile as described in the documentation you reference, I am still not able to generate the libdoc xml documentation.  I didn't initially have  that constructor in there.

I think you may be right in that I am conflating the TaskSet parameter and type.  I have found another example of locustio usage that doesn't use a constructor so I'll see if I can make this work.  It has been a struggle!

**Comment by michalanglart** [comment]
@humbienri Let's forget about RED for a moment: can you give us some small **working** example which would contain some `.robot` suite together with `.py` script in which some class inherited from `TaskSet` would serve as a Robot keywords library? From the code (https://github.com/locustio/locust/blob/master/locust/core.py) it is obvious that in order to construct `TaskSet` you need to pass another `TaskSet` or `Locust` in argument as a parent. How would you achieve that with `Library   MyTaskSet   parent` import?

**Comment by humbienri** [comment]
No I don't have a working example, even a tiny one.  That is what I am trying to figure out.  To see if it is doable at all?  But at first, I thought the problem that was tripping me up was that RED could not work with parameters in constructors.

Now I am thinking the problem lies elsewhere, or being able to pass in not s simple parameter but a parameter of type TaskSet, or Locust?

**Comment by michalanglart** [comment]
From the source code I've linked above it looks like: take a look into lines 305-310. 

Does your `TaskSet` needs to be a keyword library at all? As I said earlier - I'm not familiar with `locust` library at all so maybe this question is just stupid :). Libraries are meant to be used somehow. How would like your library `TaskSet` library to be used? What keywords would you like to have there? How should the test look like in `.robot` file?

**Comment by humbienri** [comment]
I am familiar with those lines of code, yes.  I remember now why I was trying to include an init constructor in my custom class lib, b/c I was getting that error in line 310 about the "parent having to be of type TaskSet or of type Locust".  That is what prompted me to throw in that constructor and maybe use the super() function inside of the constructor.

I am also not familiar with locust.  I am trying to see if we could incorporate it into our toolset here at work.  No, @michalanglart, your question is not stupid or dumb at all.  Trust me, I am in over my head here a little bit so if anything I think my questions may be too basic, but I am trying!  I am the "most technical" on the current team I am on.  So I don't mind your questions at all!  I thank you!

My vision was for my TaskSet library to be used in multiple places.  By performance tests using locustio, by api functional tests using either Robot Framework (RF) or PyTest, and to assist my UI tests using RF and SeleniumLibrary.

So the methods/keywords in the TaskSet library would be things like, login(), logout(), create_manual_checks_batch(), delete_general_journal_batch() and things like that.  It would basically be an API library for multi-use.  I realize I could do this without pulling locustIO into the mix...but then I wouldn't be able to leverage said API calls into a performance/load testing scenario.

Makes sense?

**Comment by humbienri** [comment]
I'll add more here to not have one post be so long winded.

I already had a working prototype for having the API calls (not the actual tests and status_code verifications) in a custom python library using the python requests module and that in turn could be used by my UI tests to do some set up functionality.  And that was working.  Again, it was just a small prototype.

But the minute I tried to pull in locustio into the mix and using inheritance...things just fell apart ;)

**Comment by michalanglart** [comment]
I've experimented a little bit and I think even from Robot code point of view it is a little problem that you need an instance of `TaskSet` or `Locust` in order to create another `TaskSet`. In terms of Robot code it means that you need to pass an instance of `TaskSet/Locust` when importing your `TaskSet`.

For example:
```
*** Settings ***
Library    MyTaskSet    ${x}
```
where `${x}` should store an instance of `TaskSet/Locust`. This mean that you would need to create one in Robot code somewhere before the `Library` import happens. Instead you could create a constructor with optional argument - `locustio` would use it by providing needed instance, while Robot would use the constructor without providing any arguments.

For example:
```python
from locust import Locust, TaskSet, task

class MyTaskSet(TaskSet):
    
    def __init__(self, locust=None):
        l = locust if locust else MyLocust()
        super(MyTaskSet, self).__init__(l)

    @task
    def task1(self):
        print(1)

    @task
    def task2(self):
        print(2)

class MyLocust(Locust):
    task_set = MyTaskSet
    min_wait = 1000
    max_wait = 2000
```

Now you can run `locustio` tests:
```
locust -f library.py
```

and moreover you can use `MyTaskSet` as a library without passing the required instance:

```
*** Settings ***
Library    library.MyTaskSet

*** Test Cases ***
test
    Task 1
    Task 2
```

Have a look at examplary project I'm attaching:
[locustProject.zip](https://github.com/nokia/RED/files/3261196/locustProject.zip)

**Comment by humbienri** [comment]
Hello again, good morning here

If this works that will be awesome!  I'll give this a try once I get into the office and report back to you.

Thank you sir!

**Comment by humbienri** [comment]
Been one of those days.  So I finally got around to trying your example and applying it to mine as you suggested.  However, I am still seeing RED getting stuck on launching my PyDev/Robot debug session.  See screenshot below:

![_2019Jun06215317](https://user-images.githubusercontent.com/1178569/59078906-db7f1c00-88a6-11e9-918b-67df66dc60c9.png)

Notice the "Launching configuration progress bar underneath.  Never moves past 24%.

Notice the highlighted Keyword, Loginloc.  It is not underscored red so RED recognizes it it seems.  When I place the cursor on that KW and hit F3, then I get the Progress dialog which also hangs up Discovering libraries.  Clicking Cancel on that dialog does nothing.  I end up having to kill RED/Eclipse from the Task Scheduler.

The \workspace\metadata\log file shows:
```
eclipse.buildId=4.9.0.I20180906-0745
java.version=1.8.0_161
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86, WS=win32, NL=en_US
Command-line arguments:  -os win32 -ws win32 -arch x86

!ENTRY org.eclipse.jface 2 0 2019-06-06 21:47:40.119
!MESSAGE Keybinding conflicts occurred.  They may interfere with normal accelerator operation.
!SUBENTRY 1 org.eclipse.jface 2 0 2019-06-06 21:47:40.119
!MESSAGE A conflict occurred for CTRL+=:
Binding(CTRL+=,
	ParameterizedCommand(Command(org.eclipse.ui.edit.text.zoomIn,Zoom In,
		Zoom in text, increase default font size for text editors,
		Category(org.eclipse.ui.category.textEditor,Text Editing,Text Editing Commands,true),
		org.eclipse.ui.internal.WorkbenchHandlerServiceHandler@19864d6,
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.eclipse.ui.textEditorScope,,,system)
Binding(CTRL+=,
	ParameterizedCommand(Command(org.robotframework.red.zoomInText,Zoom In,
		,
		Category(org.robotframework.ide.eclipse.robotCommands,Robot,null,true),
		org.eclipse.ui.internal.WorkbenchHandlerServiceHandler@12d7f34,
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.robotframework.ide.eclipse.tableeditor.sources.context,,,system)
!SUBENTRY 1 org.eclipse.jface 2 0 2019-06-06 21:47:40.119
!MESSAGE A conflict occurred for CTRL+-:
Binding(CTRL+-,
	ParameterizedCommand(Command(org.eclipse.ui.edit.text.zoomOut,Zoom Out,
		Zoom out text, decrease default font size for text editors,
		Category(org.eclipse.ui.category.textEditor,Text Editing,Text Editing Commands,true),
		org.eclipse.ui.internal.WorkbenchHandlerServiceHandler@1a7ef3e,
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.eclipse.ui.textEditorScope,,,system)
Binding(CTRL+-,
	ParameterizedCommand(Command(org.robotframework.red.zoomOutText,Zoom Out,
		,
		Category(org.robotframework.ide.eclipse.robotCommands,Robot,null,true),
		org.eclipse.ui.internal.WorkbenchHandlerServiceHandler@1cc5a38,
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.robotframework.ide.eclipse.tableeditor.sources.context,,,system)
```
which to me seems irrelevant to the problem at hand, Zoom in/out keybinding conflicts?

Also, once I rewrote the constructor for my library to mirror yours, then the importing of the library

Library    locustinger

looks fine.  No complaint about a missing argument. No red underline.  I'll add more screenshots after I kill and restart RED.

**Comment by michalanglart** [comment]
This keybinding conflict is not related. You can just ignore it

Looking at the screenshot: could you remove the entry in `red.xml` which is responsible for those two: `robot.libraries.DateTime` and `site-packages`? The first one is standard Robot library and it should be already recognized under `Robot Standard libraries` while the latter is not a keywords library in fact.

**Comment by michalanglart** [comment]
Can you also go to the source of the `Loginloc` keyword manually (without using `F3`) put a breakpoint there and try to launch debugging session? Does it work? It seems to me that the source search is the problem here.

**Comment by humbienri** [comment]
Hello,

I've gone ahead and removed the 2 references in red.xml as you had requested.  I did think it was weird that they were there since I don't remember explicitly adding them myself.  But maybe it's just my memory.

I had already previously tried to just open up my locustinger.py file and adding a breakpoint both in the constructor and in the Loginloc KW, without using F3 but my Debug session/configuration would still lock up.  

One thing I did notice was that if I remove the call to python.exe from my debug configuration, then my debug session would NOT hang up and run along just fine but none of my breakpoints in locustinger.py would be hit.  See screenshot:
![_2019Jun07084852](https://user-images.githubusercontent.com/1178569/59108981-c71c3d00-8901-11e9-9b2a-3a811020daec.png)

Just an observation.

In your sample locustProject, I noticed that in your tasks methods, you did not actually use the l object.  Also, you were using the Locust object rather than the HttpLocust object, which I do think is the one that I'll be needing since I plan to use this lib for making API calls.

Thank you for your help @michalanglart .

**Comment by michalanglart** [comment]
> One thing I did notice was that if I remove the call to python.exe from my debug configuration, then my debug session would NOT hang up and run along just fine but none of my breakpoints in locustinger.py would be hit

Can you try creating another project, write simple test:
```
*** Test Cases ***
test
    Log    10
```
put a breakpoint in `Log 10 line` then hit F3 to go to the source of `Log` keyword put a breakpoint there and try to arrange RED-PyDev session and see if it works?

> you did not actually use the l object

How come? I'm passing this object to constructor of `TaskSet`:

```python
class MyTaskSet(TaskSet):
    def __init__(self, locust=None):
        l = locust if locust else MyLocust()
        super(MyTaskSet, self).__init__(l)
```

In the first line of constructor I'm assigning `locust` to variable `l` if `locust` is other than `None` otherwise I'm creating new object of type `MyLocust`. Then in the second line the object assigned to `l` is passed to constructor of super class - `TaskSet`.

>Also, you were using the Locust object rather than the HttpLocust object

Well this was an example to make it as simple as possible. Your `Locust` class can freely extend `HttpLocust` if you wish. It doesn't change anything in task set class.

**Comment by humbienri** [comment]
Sorry, I meant that you then didn't use the l object within your task methods.  This is what I have so far for my locustinger.py:
```
from locust import Locust, HttpLocust, TaskSet, task

class locustinger(TaskSet):
    ROBOT_LIBRARY_SCOPE = 'GLOBAL'
    
    def __init__(self, locust=None):
        self.l = HttpLocust if locust else WebsiteUser()
        super(locustinger, self).__init__(self.l)
 
    @task
    def loginloc(self):
        login_creds = {
            'user' : 'xxxxx',
            'password' : 'xxxxxxx',
            'org' : 'xxxxxx'
        }

        r = self.l.client.post("https://dev-api.compeat.com/v1/login", json=login_creds, verify=False)
        print("Oh hay!")
#         return "Yeppers"
        return r

class WebsiteUser(Locust):
    task_set = locustinger
    host = 'https://devl-api.compeat.com'
    min_wait = 5000
    max_wait = 9000
```

**Comment by humbienri** [comment]
Yesterday I had also tried to apply my debug configuration to the sample project that you provided.

Appy a breakpoint on line 6, Task 1 KW, hit F3 and it does take me to the source file for that lib.  So it is aware of the source code when navigating and doing edits.

But when I try to my debug configuration on it, then RED just hangs.  Again at 24%, Launching Configuration, Launching delegate...
![_2019Jun07095056](https://user-images.githubusercontent.com/1178569/59112981-cd161c00-8909-11e9-91c2-0f5617cd7eb4.png)

**Comment by humbienri** [comment]
Suspecting that maybe there is something odd/wrong/corrupt about the settings applied and/or autogenerated by my debugging configuration, I went through the steps again to generate another separate debug configuration as explained here:  http://nokia.github.io/RED/help/user_guide/launching/debug/robot_python_debug.html

and used that debug configuration on your sample project you provided me, locustProject.

The end result behavior appears to be the same, hanging on launching delegate.  But if you look at the screenshow below of the Console output, does that look correct to you?  Particularly the generated commandline python.exe command? Also, notice how there is an error about not being able to import the library `library.MyTaskSet`...No module found error.
```
![_2019Jun07125640](https://user-images.githubusercontent.com/1178569/59133071-81309a80-893c-11e9-9a20-a7e1eab3b906.png)
```

So, F3 source code navigating finds the source code but not while debugging.

**Comment by michalanglart** [comment]
As to the `l` object: you don't need to store it in `self.l`. In my example `l` is passed to the constructor of super class `TaskSet` which will put this object into `locust` field as visible here: https://github.com/locustio/locust/blob/84ffbae085268b62574f726fcb01a18b0f908e7b/locust/core.py#L308 This mean that in the inheriting class you can simply use `self.locust` or even self.client`.

Regarding the error about the module: I'm able to reproduce that and it looks like a regression from 0.8.11. We need to have a deeper look. I was unable to reproduce the "hanged" launching delegate job however.

I'll get back to that on monday.

**Comment by humbienri** [comment]
Thank you sir!  I will continue to beat my head at this over the weekend. Have a great weekend yourself.

Thank you very much and very sincerely.

What part of the world are you in?

**Comment by michalanglart** [comment]
RED team is part of Nokia in Poland, currently in CEST time.

Have a great weekend, don't beat your head too much :)

Also: if you have time, you may look if you're able to configure all of that with RED 0.8.11. Take into account however that between 0.8.11 and 0.8.12, we've changed:

1. the way libraries paths are written to red.xml, so it needs to be created from scratch
2. the way RED & PyDev debugging session works; it was a bit more complicated and unclear (although described in Help)

I'm suspecting that the change (1) is a possible cause of the importing problem we're both seeing in the `Console` view.

**Comment by michalanglart** [comment]
I'm able to configure the hanging behavior even on my example project. I don't know however what is causing it. It seems that the whole tests execution just hangs but only once running through `redpydevd`/`pydevd` debugger.

**Comment by michalanglart** [comment]
Ok, so it looks like I finally figured it out. 

I've created a library totally unrelated to locust with the single except that it did imported `locust` module (although not used):

```python
import locust

class Library(object):        
    def keyword(self):
        print(1)
```

And it was also hanging. Then I've created simple python script:
```python
print(1)
import locust
print(2)
```
and tried to debug it from eclipse with PyDev. It was also hanging on `locust` import. To be more specific: `locust` was importing `locust.core` which then was importing `locust.events` which was importing `gevent` and this was the cause. 

I've googled for that and this is known issue and there is a solution for that: in both Eclipse with PyDev and IntelliJ/PyCharm (they are both using `pydevd` module for debugging) there are preferences to be checked in order to support `gevent`. In terms of `pydevd` this translates to `GEVENT_SUPPORT` environment variable being set to `True`. 

This mean that you can fix this issue by defining this variable in launch configuration as visible on screenshot below. This solved the issue on my side and I was able to normally debug with RED/PyDev. Can you confirm on your side?

![gevent](https://user-images.githubusercontent.com/14932410/59200631-f698a700-8b98-11e9-94e0-624f20f5fdf8.png)

I think we would also add similar checkbox for `gevent` support in the launch config wizard.

**Comment by humbienri** [comment]
Thank you so very much @michalanglart.  You are definitely going above and beyond what is expected from you.  Certainly by me.

Owe you a beer!  :)

So we have progress now! Once I set those 2 Environment variables, I am able to hit my breakpoints in my custom lib.  A big thank you!

Unfortunately, I am now reaching other errors that I think are not related to this problem.  I'm getting a StackOverflow error that seems to be related to `gevent`.

Over the weekend I did try reverting back to v. 0.8.11 briefly.  I noticed one thing, even on v. 0.8.11...I would put my cursor on my `Loginloc` keyword in my .robot file and hit `F3` and RED would just hand up just like it did when on v. 0.8.12.  It would get hung up on `Discovering library: SeleniumLibrary` I would do nothing else than that and would after quite a few minutes would resort to force killing Eclipse from the Task Manager (I'm on Windows).

I then re-upgraded to v. 0.8.12 of RED and try the same experiment and RED would still just hang up as well.

Now, after having applied your "Environment Variables solution" I repeat the exercise and RED still hangs up discovering libraries.  Again, just an `F3` with the cursor on my Keyword of interest (custom lib).

If I repeat this with the sample `locustProject` you provided me, same result.  It gets hung up on `Disvovering library: Screenshot`

Today, I am also seeing quite a few of those `XPC` calls.  I've enabled RED to show/report more errors when they happen.

**Comment by michalanglart** [comment]
> Owe you a beer! :)

The cold one please! Given the temperatures we're having now in Wrocław... :)

> Once I set those 2 Environment variables

You need `GEVENT_SUPPORT` only. The `PYTHONIOENCODING` is added by RED to all launch configurations by default (although it's configurable in Preferences) in order to properly handle non-ASCII characters from chinese alphabets, cyryllic etc.

> Unfortunately, I am now reaching other errors that I think are not related to this problem. I'm getting a StackOverflow error that seems to be related to gevent.

I haven't seen anything like that in my case. I've seen this hanging on F3 however just before leaving the office. Overall there are 3 problems:

1) Execution hangs on `gevent` import. This can be fixed by setting `GEVENT_SUPPORT` environment variable to `True`

2) Keyword source searching hangs. I don't know yet why although it could be `gevent` too however of course `GEVENT_SUPPORT` will not work here as this is a parameter for `pydevd` module which has nothing to do with keyword searching

3) There is a regression in generating command line arguments: the python path may be wrongly generated for libraries having name like `module.class`. This was mentioned here: https://github.com/nokia/RED/issues/308#issuecomment-500035481 and confirmed by me a comment later. This however has nothing to do with all those hanging issues.

**Comment by michalanglart** [comment]
I've investigated the keyword searching mechanism. As suspected `gevent` is a cause here too. Unfortunately this is needed to be fixed on our side and it will need to be done in next release. Probably we will introduce a preference too similarly as PyDev does it.

**Comment by humbienri** [comment]
Hello @michalanglart,

Sorry, I meant to get back to you sooner.  I've gone ahead and did as you suggested, dropped usage of the `PYTHONIOENCODING` environmental variable and only added the `GEVENT_SUPPORT` variable.

Glad to hear that you are getting to the bottom of this.

In the meantime I'll continue to "hobble along" with my loucst pydevd task.  It is kicking my butt!

How hot is it over there?  Here, this past weekend was miserably hot.  And humid!!  Quite nasty for being outside!

If I am ever in Wroclaw, you're on! Beer time!  =)

**Comment by michalanglart** [comment]
> In the meantime I'll continue to "hobble along" with my loucst pydevd task. It is kicking my butt!

But it looks that everything will work with next version of RED! 

> How hot is it over there?

It was 34 degrees C today and around 20 in the night. Nothing unusual for this part of the world but anyway you don't have too many such days troughout the whole year. And where are you located by the way?

**Comment by humbienri** [comment]
Fingers crossed!  I certainly hope so.  RED is great...but this would make it greater! Awesome even!

I'm in Austin, TX.  It's been hot and humid.  Around this time of year, it is normal for it to be hot, but not as humid.  Humid just makes it worse! We're nearing the upper 90s, Farenheit.

Although, this past Sunday we had some ugly ugly rainstorms with very strong winds.  Some roofs were damaged.  Followed by thunder and lightning.  That's kind of unusual and hopefully not the new norm.

**Comment by michalanglart** [comment]
Sorry it took that long but we've finally released 0.8.13 with fixes for this issue

**Comment by humbienri** [comment]
@michalanglart Thank you sir!  It is sir right?

No problem at all.  I'm grateful indeed! Still loving RED and definitely looking forward to upgrading to 0.8.13.  Scared too! I think I'll wait 'til Monday though...just in case :)

Best Wishes!
-Humberto

**Comment by michalanglart** [comment]
> Thank you sir! It is sir right?

Yes, that's right

Let us know if it works for you with 0.8.13
